test/functional/provider/clipboard_spec.lua

```diff
@@ -752,4 +752,11 @@ describe('clipboard (with fake clipboard.vim)', function()
     expect('some some')
     eq('some', eval('getreg("*")'))
   end)
+
+  it('has correct value for clipboard registers #24257', function ()
+    eval('setreg("*", "")')
+    eval('setreg("+", "waddup")')
+    eq("", eval('getreg("*")'))
+    eq("waddup", eval('getreg("+")'))
```

Comment:
On a non-X11 system where there is no distinction between `"*` and `"+"`, those two registers are linked; so aren't they supposed to return the same register result `"waddup"`? How can the test pass?

Reply:
That's a great callout! Correct me if I'm wrong, but I don't think they're _necessarily_ linked, just linked by default. a user in MacOS could technically override `g:clipboard` to set them to do whatever they want. For example, `*` could be configured to use the tmux clipboard while `+` uses the system clipboard (source: [provider docs](https://neovim.io/doc/user/provider.html)).

I think the easiest way to avoid non-X11 weirdness in this test is to disable to clipboard provider, ensuring that `get_clipboard` will fail, then running this test.